modules/statistics-provider-matomo/src/main/java/org/opencastproject/statistics/provider/matomo/StatisticsProviderMatomoService.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ *
+ * The Apereo Foundation licenses this file to you under the Educational
+ * Community License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License
+ * at:
+ *
+ *   http://opensource.org/licenses/ecl2.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package org.opencastproject.statistics.provider.matomo;
+
+import org.opencastproject.statistics.api.StatisticsCoordinator;
+import org.opencastproject.statistics.api.StatisticsProvider;
+import org.opencastproject.statistics.provider.matomo.provider.BatchMatomoRequest;
+import org.opencastproject.statistics.provider.matomo.provider.MatomoProviderConfiguration;
+import org.opencastproject.statistics.provider.matomo.provider.MatomoTimeSeriesStatisticsProvider;
+import org.opencastproject.util.ConfigurationException;
+
+import org.apache.felix.fileinstall.ArtifactInstaller;
+import org.osgi.service.cm.ManagedService;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.Dictionary;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Implements statistics providers using Matomo for data retrieval.
+ */
+@Component(
+    immediate = true,
+    service = { ManagedService.class, ArtifactInstaller.class },
+    property = {
+        "service.description=Statistics Provider Matomo Service"
+    }
+)
+public class StatisticsProviderMatomoService implements ManagedService, ArtifactInstaller {
```

Comment:
Do not use ManagedService, that should not be necessary.
```suggestion
@Component(
    immediate = true,
    service = { ArtifactInstaller.class },
    property = {
        "service.description=Statistics Provider Matomo Service"
    }
)
public class StatisticsProviderMatomoService implements ArtifactInstaller {
```

Reply:
Thanks for pointing that out! I wasn't aware of that and did copy this part from the InfluxDB statistics provider.